### DISCUSSION TITLE
Files with a "%" in the file name cannot be resolved

### DISCUSSION THREAD

**Discussion by FrazeColder** [spatie/laravel-medialibrary#2201] [Ideas]
Hi,

I found a bug. When uploading an image with an `%` in the file name e.g. "50% discount.jpg" the image cannot be resolved. This is due to a `%` is a special char. The bug doesn't come from spatie media library itself but you guys could fix this bug by replacing all `%` within filenames. 

I guess this would be a great add to prevent problems for other users!

Kind regards

**Comment by lukasleitsch** [comment]
@FrazeColder Did you find a solution for this issue? We have the same encoding issue. The user uploads a file that contains `%` or `&`. The output of the URL looks like this:

```html
<a href="{{ $media->getUrl() }}">Download</a>
```

The browser can't access the file due to the escaping of the URL.

Adding `download` to the `a` tag could solve it and the browser can load the file. But this doesn't solve it for using the url in an image tag. The browser can't render the image if the url contains `%` or `&`.

Another solution could be to use a custom `FileNamer`. In this FileNamer, we could use the Laravel `Str::slug()` Helper. But this would not solve the issue for already uploaded files.

https://spatie.be/docs/laravel-medialibrary/v11/advanced-usage/naming-files
